clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -313,46 +296,64 @@ basic_counts_t::get_total_counts()
     return total;
 }
 
-void
-basic_counts_t::compute_shard_interval_result(per_shard_t *shard, uint64_t interval_id)
+analysis_tool_t::interval_state_snapshot_t *
+basic_counts_t::generate_shard_interval_snapshot(void *shard_data, uint64_t interval_id)
 {
-    counters_t shard_total;
-    for (const auto &ctr : shard->counters) {
-        shard_total += ctr;
+    per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+    counters_t cur_counters;
+    for (const auto &ctr : per_shard->counters) {
+        cur_counters += ctr;
```

Comment:
> presumably leading to huge empty series of wall-clock-based intervals in each gap

There won't be any empty intervals in the gap. Interval snapshots are created at timestamp markers, so if the next window is farther than `-interval_microseconds` the next window will fall into the next interval. The only gap we'll see is in the interval_id and interval_end_timestamp.

>  would expect a separate time series within each window

I think that'll be useful to add in a structured output, but it may make the stderr output too big. I added a TODO for now. Would like to avoid making this PR any bigger.